dev/io.openliberty.checkpoint.openj9/src/io/openliberty/checkpoint/openj9/internal/ExecuteCRIU_OpenJ9.java

```diff
@@ -42,17 +42,51 @@ public void stop(BundleContext bc) {
     }
 
     @Override
-    public int dump(File directory) throws IOException {
+    public SnapshotResult dump(File directory) {
+        SnapshotResult snapshotResult = new SnapshotResult(SnapshotResultType.SNAPSHOT_FAILED, "The criu dump command failed with error: Snapshot Failed", null);
+
         if (!CRIUSupport.isCRIUSupportEnabled()) {
             // TODO log appropriate message
             System.out.println("Must set the JVM option: -XX:+EnableCRIUSupport");
-            return -50;
+            return snapshotResult;
         }
         CRIUResult result = CRIUSupport.checkPointJVM(directory.toPath());
-        if (result.getType() == CRIUResultType.SUCCESS) {
-            return 1;
+
+        switch (result.getType()) {
+            case SUCCESS:
+                snapshotResult = createSnapshotResult(SnapshotResultType.SUCCESS, "Success");
+                break;
+
+            case UNSUPPORTED_OPERATION:
+                snapshotResult = createSnapshotResult(SnapshotResultType.UNSUPPORTED_OPERATION, "The criu dump command failed with error: Unsupported Operation");
+
+            case INVALID_ARGUMENTS:
+                snapshotResult = createSnapshotResult(SnapshotResultType.INVALID_ARGUMENTS, "The criu dump command failed with error: Invalid Arguments");
+                break;
+
+            case SYSTEM_CHECKPOINT_FAILURE:
+                snapshotResult = createSnapshotResult(SnapshotResultType.SYSTEM_CHECKPOINT_FAILURE, "The criu dump command failed with error: System Checkpoint Failure");
```

Comment:
Here I expect j9 will get some updates to give us access to the error code from `criu dump` operation